packages/core/test/ServerTest.ts

```diff
@@ -116,13 +155,161 @@ class WoTServerTest {
             value: 10
         };
         thing.addProperty(initp);
-        let value1 = await thing.readProperty("number");
-        expect(value1).to.equal(10);
-        await thing.writeProperty("number", 5);
+        let value0 = await thing.readProperty("number");
+        expect(value0).to.equal(10);
+        let value1 = await thing.writeProperty("number", 5);
```

Comment:
writeProperty() must not return the written value. It only returns a success status (no value) or error response (maybe a value).

Reply:
There can be such a behavior for the ExposedThing as added feature by the implementation, but not on the ConsumedThing, as the defaults for HTTP and CoAP do not allow to reply with a state representation.

Furthermore, it is invalid to ultimately write a different state than intended by the client (and treat it as one transaction). See PUT semantics. The writeHandler is for additional side effects.

Please fix spec.